corehq/tests/test_middleware.py

```diff
@@ -173,23 +173,19 @@ class TestLogLongRequestMiddlewareReports(TestCase):
     def setUpClass(cls):
         super().setUpClass()
         cls.domain = Domain(name="long_request", is_active=True)
+        cls.addClassCleanup(cls.domain.delete)
         cls.domain.save()
 
         cls.username = 'fingile'
         cls.password = '*******'
         cls.user = WebUser.create(cls.domain.name, cls.username, cls.password, None, None)
+        cls.addClassCleanup(cls.user.delete, cls.domain.name, deleted_by=None)
```

Comment:
In my tests, should I be using `add(Class)?Cleanup` instead of `tearDown(Class)?`?

Reply:
Yes, it's more reliable. If the setup method crashes all cleanups registered up to that point will be called in reverse order (teardowns would not be called in that case). Also, if a cleanup crashes it does not prevent other cleanups from running.